### ISSUE TITLE
[bug] bud-swc and the tree shake murder mystery

### ISSUE THREAD

**Issue by joshuafredrickson** [roots/bud#2088]
### Agreement

- [X] This is not a duplicate of an existing issue
- [X] I have read the [guidelines for Contributing to Roots Projects](https://github.com/roots/.github/blob/master/CONTRIBUTING.md)
- [X] This is not a personal support request that should be posted on the [Roots Discourse](https://discourse.roots.io/) community

### Describe the issue

In versions 6.6.6+, `bud-swc` breaks tree-shaking in `@fortawesome/fontawesome-svg-core`.

Reference: [Discord thread](https://discord.com/channels/461985401494700032/1072261001929556068)

Many many thanks!

### Expected Behavior

The final build should only output imported icons, resulting in a smaller build size.

```
◉  sage ./dist [6097681e832a6dd74bae]
│
├─ entrypoints
│ └─ main
│   └─ js/main.js    61.82 kB
│
│
└─ compiled 3 modules in 1s 912ms
```

### Actual Behavior

The final build includes all the icons, resulting in a larger build size.

```
◉  sage ./dist [35c3a4bc349ef0c985cb]
│
├─ entrypoints
│ └─ main
│   └─ js/main.js     1.03 MB
│
│
└─ compiled 6 modules in 4s 53ms
```

### Steps To Reproduce

To quote @kellymears :

> These were the swc options in 6.5.4 https://github.com/roots/bud/blob/69d85b104bd93730376875a1598394e8a27910fb/sources/%40roots/bud-swc/src/extension.ts#L20-L35
> These are the current options: https://github.com/roots/bud/blob/10bd917139bc09f1a8a35ecc270204277a4541d5/sources/%40roots/bud-swc/src/extension/index.ts#L19-L31
> 
> **[Forked the sandbox with a fix.](https://codesandbox.io/p/sandbox/upbeat-heyrovsky-0vxh19?file=%2Fbud.config.js&selection=%5B%7B%22endColumn%22%3A1%2C%22endLineNumber%22%3A13%2C%22startColumn%22%3A1%2C%22startLineNumber%22%3A13%7D%5D)** Did it as an extension so it's kinda hopefully portable.
> 
> Related issue:
https://github.com/roots/bud/blob/10bd917139bc09f1a8a35ecc270204277a4541d5/sources/%40roots/bud-swc/src/extension/index.ts#L118-L121
> Doesn't respect extension options. this should have been as easy as:
```bud.swc.set(`module`, undefined)```

### version

6.9.0

**Comment by kellymears** [comment]
Okay, seems like the fix was successful so I'll release a patch. 

Sandbox of `nightly@2023.2.8`: https://codesandbox.io/p/sandbox/bud-2023-2-8-iphbh0

**Comment by kellymears** [comment]
Released in [6.9.1](https://bud.js.org/releases/6.9.1)!
